game-core/src/main/java/games/strategy/triplea/ui/menubar/TripleAMenuBar.java

```diff
@@ -60,60 +56,19 @@ private void createLobbyMenu(final InGameLobbyWatcherWrapper watcher) {
    *     cancelled the operation.
    */
   public static File getSaveGameLocation(final Frame frame) {
-    // For some strange reason, the only way to get a Mac OS X native-style file dialog
-    // is to use an AWT FileDialog instead of a Swing JDialog
-    if (SystemProperties.isMac()) {
-      final FileDialog fileDialog = new FileDialog(frame);
-      fileDialog.setMode(FileDialog.SAVE);
-      fileDialog.setDirectory(ClientSetting.saveGamesFolderPath.getValueOrThrow().toString());
-      fileDialog.setFilenameFilter((dir, name) -> GameDataFileUtils.isCandidateFileName(name));
-      fileDialog.setVisible(true);
+    final FileDialog fileDialog = new FileDialog(frame);
+    fileDialog.setMode(FileDialog.SAVE);
+    fileDialog.setDirectory(ClientSetting.saveGamesFolderPath.getValueOrThrow().toString());
+    fileDialog.setFilenameFilter((dir, name) -> GameDataFileUtils.isCandidateFileName(name));
+    fileDialog.setVisible(true);
 
-      final String fileName = fileDialog.getFile();
-      if (fileName == null) {
-        return null;
-      }
-
-      // If the user selects a filename that already exists,
-      // the AWT Dialog on Mac OS X will ask the user for confirmation
-      return new File(fileDialog.getDirectory(), GameDataFileUtils.addExtensionIfAbsent(fileName));
-    }
-
-    // Non-Mac platforms should use the normal Swing JFileChooser
-    final JFileChooser fileChooser = SaveGameFileChooser.getInstance();
-    final int selectedOption = fileChooser.showSaveDialog(frame);
-    if (selectedOption != JFileChooser.APPROVE_OPTION) {
+    final String fileName = fileDialog.getFile();
+    if (fileName == null) {
       return null;
     }
-    File f = fileChooser.getSelectedFile();
-    // disallow sub directories to be entered (in the form directory/name) for Windows boxes
-    if (SystemProperties.isWindows()) {
-      final int slashIndex = Math.min(f.getPath().lastIndexOf("\\"), f.getPath().length());
-      final String filePath = f.getPath().substring(0, slashIndex);
-      if (!fileChooser.getCurrentDirectory().toString().equals(filePath)) {
-        JOptionPane.showConfirmDialog(
-            frame,
-            "Sub directories are not allowed in the file name.  Please rename it.",
-            "Cancel?",
-            JOptionPane.DEFAULT_OPTION,
-            JOptionPane.WARNING_MESSAGE);
-        return null;
-      }
-    }
-    f = new File(f.getParent(), GameDataFileUtils.addExtensionIfAbsent(f.getName()));
-    // A small warning so users will not over-write a file
-    if (f.exists()) {
```

Comment:
Hmm I believe we should have that warning on all systems

Reply:
It is at least for sure on linux, it seems to be an OS default baked in already :+1:

Reply:
Good to know, just double checked if this is the case on windows as well and it is 👍